scripts/lease.sh

```diff
@@ -2,24 +2,35 @@
 
 lease() {
   local tries=1
+  local cluster_args
+
+  # place priority over cluster regex (name)
+  if [ -n "${CLUSTER_REGEX}" ]; then
+    cluster_args="--cluster-regex=${CLUSTER_REGEX} "
+  elif [ -n "${CLUSTER_VERSION}" ]; then
+    cluster_args="--cluster-version=${CLUSTER_VERSION} "
+  fi
+  cluster_args="${cluster_args}--duration=${CLUSTER_DURATION}"
+
+  echo "Requesting lease with: '${cluster_args}'"
   while [ -z "$TOKEN" ]; do
-    eval "$(k8s-claimer --server=k8s-claimer-e2e.deis.com lease create --duration="${CLUSTER_DURATION}")"
+    eval "$(k8s-claimer --server=k8s-claimer-e2e.deis.com lease create "${cluster_args}")"
```

Comment:
@jchauncey ftw! I indeed think this was messing things up (quotes added b/c of a shellcheck failure but apparently the advice is ill-advised in this case)